pkg/sbom/cyclonedx/marshal.go

```diff
@@ -428,16 +478,22 @@ func (e *Marshaler) resultToCdxComponent(r types.Result, osFound *ftypes.OS) cdx
 	case types.ClassOSPkg:
 		// UUID needs to be generated since Operating System Component cannot generate PURL.
 		// https://cyclonedx.org/use-cases/#known-vulnerabilities
-		component.BOMRef = e.newUUID().String()
+		// component.BOMRef = e.newUUID().String()
+		// custom logic instead of UUID for consistent report
+		component.BOMRef = bomRef(string(r.Class))
```

Comment:
Where does reference show up? Do you have a sample?

Reply:
```
"components": [
    {
      "bom-ref": "os-pkgs/debian/12.1",
      "type": "operating-system",
      "name": "debian",
      "version": "12.1",
      "properties": [
```

Reply:
Is this a sample for ` bomRef(string(r.Class), osFound.Family, osFound.Name)` or ` bomRef(string(r.Class))`

Reply:
` bomRef(string(r.Class), osFound.Family, osFound.Name)`

`bomRef(string(r.Class))` this is applicable when there is no os information. Currently I don't have a sample image for that

If there is no OS info, the bom-ref will be `os-pkgs`
```
"components": [
    {
      "bom-ref": "os-pkgs",
```

Reply:
Oh ok